crates/biome_deserialize/src/js.rs

```diff
@@ -0,0 +1,68 @@
+//! Implementation of [DeserializableValue] for the JS engine.
+use crate::{DeserializationDiagnostic, Deserialized};
+use biome_console::markup;
+use biome_diagnostics::Error;
+use boa_engine::{Context, JsValue};
+use serde::de::DeserializeOwned;
+
+/// Attempts to interpret a value from the JS runtime and deserialize it following the semantics
+/// defined by JSON. As such, JS types that cannot be serialized to JSON, such as functions or
+/// Promises are not supported by this deserializer.
+///
+/// The data structures that need to be deserialized have to implement the [Deserializable] trait.
+/// For most data structures, this can be achieved using the
+/// [biome_deserialize_macros::Deserializable] derive.
```

Comment:
Note this comment is not entirely correct, since the function uses Serde traits for now, instead of our `Deserializable`. If we agree on picking Boa, I would optimize this function to use `Deserializable` again so we can go straight from `boa_engine::JsValue` to our own types, instead of using `serde_json::Value` as intermediary.